snapcraft/plugins/jhbuild.py

```diff
@@ -0,0 +1,1035 @@
+  - modules (required): a list of modules to include in the part, modules that
+                        should be skipped should be prefixed with '-'
+  - moduleset: the module set JHBuild should use (default: gnome-world)
+  - snap-name (required): the name of the snap this part is included in
+  - snap-revision: the revision of the snap this part is included in
+                   (default: current)
+  - container-name: the name of the LXC container used for building
+                    (default: snapcraft-<snap name>-<part name>)
+  - distribution: the distribution the LXC container is based off of
+                  (default: host distribution)
+  - release: the release of the distribution the LXC container is based off of
+             (default: host release)
+  - architecture: the architecture of the LXC container
+                  (default: host architecture)
+  - container-packages: a list of build packages the LXC container should be
+                        bootstrapped with
+  - debmirror: the mirror to use for downloading .deb packages
+  - jhtarballs: the directory on the host containing source tarballs
+  - jhmirror: the directory on the host containing DVCS mirror repositories
+  - jhcheckout: the directory on the host container DVCS checkout repositories
+  - ccache: the directory on the host to use for ccache caching
+  - disable-pull: True to skip the pull stage (default: False)
+  - disable-build: True to skip the build stage (default: False)
+  - disable-jhupdate: True to skip updating jhtarballs, jhmirror and jhcheckout
+                      (default: False)
+  - disable-parallel: True to disable parallel building (default: False)
+
+Advice:
+
+  - The command for launching an app from a JHBuild part is:
+    jhbuild -f $SNAP/etc/jhbuildrc run <app>
+
+    So for a glade snap, the app would look like:
+
+    apps:
+      glade:
+        command: jhbuild -f $SNAP/etc/jhbuildrc run glade
+        plugs: [x11]
+
+  - Building WebKit requires a lot of time and computing power. If WebKit is a
+    dependency of your JHBuild module, it can be skipped by adding '-WebKit' to
+    the list of modules, and adding 'libwebkit2gtk-4.0-dev' to the list of
+    container-packages.
+
+  - Using the debmirror option in combination with a local apt-cacher-ng
+    installation can prevent repeated downloading of .deb packages. Use the IP
+    address of your lxcbr0 interface with the port number apt-cacher-ng uses.
+    For example, debmirror: http://10.0.3.1:3142
+
+  - Specify directories on your host for jhtarballs and jhmirror to prevent
+    repeated downloading of the JHBuild module sources. It is best to reserve
+    directories on your local machine that can be used for all snaps you might
+    want to build.
+
+  - If you are debugging your snap, but don't need to rebuild the sources (for
+    example, to determine the list of plugs required under confinement), use
+    the disable-pull, disable-build, and disable-jhupdate options.
```

Comment:
Ah, I see. But the problem here IMO is that if you want to debug your snap, you usually do it outside of the lxc cleanbuild. So you run pull once, and the next time it will be skipped.
I think we need some cool design to allow us to keep and reuse a lxc machine for cleanbuild, so we could do something similar.